diff -ruN gcc-4.6.1/gcc/config/linux.h ../../../android-ndk-r6/src/gcc/gcc-4.6.1/gcc/config/linux.h
--- gcc-4.6.1/gcc/config/linux.h	2011-01-12 06:29:14.000000000 -0800
+++ ../../../android-ndk-r6/src/gcc/gcc-4.6.1/gcc/config/linux.h	2011-07-27 06:06:03.469625888 -0700
@@ -96,4 +96,4 @@
 #define TARGET_C99_FUNCTIONS (OPTION_GLIBC)
 
 /* Whether we have sincos that follows the GNU extension.  */
-#define TARGET_HAS_SINCOS (OPTION_GLIBC || OPTION_BIONIC)
+#define TARGET_HAS_SINCOS (OPTION_GLIBC)
diff -ruN gcc-4.6.1/libiberty/getpagesize.c ../../../android-ndk-r6/src/gcc/gcc-4.6.1/libiberty/getpagesize.c
--- gcc-4.6.1/libiberty/getpagesize.c	2005-03-27 07:31:13.000000000 -0800
+++ ../../../android-ndk-r6/src/gcc/gcc-4.6.1/libiberty/getpagesize.c	2011-07-26 03:03:39.641152999 -0700
@@ -60,11 +60,13 @@
 # endif /* PAGESIZE */
 #endif /* GNU_OUR_PAGESIZE */
 
+#if DEFAULT_LIBC != LIBC_BIONIC
 int
 getpagesize (void)
 {
   return (GNU_OUR_PAGESIZE);
 }
+#endif
 
 #else /* VMS */
 
diff -ruN gcc-4.6.1/libiberty/setproctitle.c ../../../android-ndk-r6/src/gcc/gcc-4.6.1/libiberty/setproctitle.c
--- gcc-4.6.1/libiberty/setproctitle.c	2011-02-02 23:23:20.000000000 -0800
+++ ../../../android-ndk-r6/src/gcc/gcc-4.6.1/libiberty/setproctitle.c	2011-07-26 06:12:45.866386786 -0700
@@ -43,6 +43,6 @@
 #ifdef PR_SET_NAME
   /* On Linux this sets the top visible "comm", but not necessarily
      the name visible in ps. */
-  prctl (PR_SET_NAME, name);
+  prctl (PR_SET_NAME, name, 0, 0, 0);
 #endif
 }
